### DISCUSSION TITLE
Locale with catch-all routes returns first segment of url and not the locale

### DISCUSSION THREAD

**Discussion by jderboven** [amannn/next-intl#894] [Q&A]
### Description

When I try to enter this kind of url : http://localhost:3000/test/test2/test3/test4/test5, my locale is not the one I have in my configuration 

```
my middleware contains this one : 
import createMiddleware from 'next-intl/middleware';
import { locales, localePrefix } from './navigation';

export default createMiddleware({
  locales,
  defaultLocale: 'fr',
  localePrefix
});

export const config = {
  matcher: ['/', '/(fr|nl)/:path*']
};
```
Navigation.ts
```
import {
  createLocalizedPathnamesNavigation
} from 'next-intl/navigation';

export const locales = ['fr', 'nl'] as const;
export const localePrefix = 'never'; // Default

const pathnames = {
  '/': '/',
};

export const {Link, redirect, usePathname, useRouter, getPathname} =
  createLocalizedPathnamesNavigation({locales, localePrefix, pathnames});
```

and finally my hierarchy follows this path
<img width="236" alt="Capture d’écran 2024-02-27 à 18 05 00" src="https://github.com/amannn/next-intl/assets/13959327/e3c100d2-d8e6-4a0e-b153-4bc9354bbaee">

Am I missing somewhere ? 

### Mandatory reproduction URL

App router

### Reproduction description

Just launch the app, getting components props gives "test" as locale 

### Expected behaviour

Locale should be 'fr' or 'nl'

**Comment by amannn** [comment]
Your matcher is off. Please carefully go over the docs for `localePrefix: 'never'`.

I’ll move this to a discussion since it’s a usage question.

**Reply by jderboven** [reply]
```
import createMiddleware from 'next-intl/middleware';
import { locales, localePrefix } from './navigation';

export default createMiddleware({
  locales,
  defaultLocale: 'fr',
  localePrefix
});

export const config = {
  matcher: ['/', '/(fr|nl)/:path*']
};
```

Matcher seems ok ?

**Reply by amannn** [reply]
Nope: https://next-intl-docs.vercel.app/docs/routing/middleware#locale-prefix-never
